exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -696,6 +780,18 @@ public void executeBuildPhase() throws SchemaChangeException {
         if ( cycleNum > 0 ) {
           read_right_HV_vector = (IntVector) buildBatch.getContainer().getLast();
         }
+        //create runtime filter
+        if (cycleNum == 0 && enableRuntimeFilter) {
+          //create runtime filter and send out async
+          int condFieldIndex = 0;
+          for (BloomFilter bloomFilter : bloomFilters) {
+            for (int ind = 0; ind < currentRecordCount; ind++) {
+              long hashCode = hash64.hash64Code(ind, 0, condFieldIndex);
+              bloomFilter.insert(hashCode);
```

Comment:
Currently, the hash join relies on memory calculations (both build and probe sides) to do accounting for spilling purposes.  The bloom filter memory use should also be included in that calculation, although it would fine if you create an enhancement JIRA and address it separately.

Reply:
I will enhance this at another JIRA.